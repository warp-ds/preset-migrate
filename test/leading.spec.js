import { setup } from "./_helpers.js";
import { describe, expect, test, vi } from "vitest";

setup();


describe("leading", () => {
  test("Emits a warning if used with numbers", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = ["leading-1", "leading-12", "leading-14", "leading-16", "leading-20", "leading-24", "leading-36"]

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });

  test("Emits a warning if used with pseudo", async (t) => {
    const warnSpy = vi.spyOn(global.console, 'warn')

    const classes = ["md:text-primary", "active:text-secondary", "!text-danger"]

    const { css } = await t.uno.generate(classes);

    expect(css).toMatchInlineSnapshot('""');
    expect(warnSpy).toHaveBeenCalledTimes(classes.length);
    expect(warnSpy.calls.flat()).toMatchSnapshot();
  });
})