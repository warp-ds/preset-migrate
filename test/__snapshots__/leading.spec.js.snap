// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`leading > Emits a warning if used with numbers 2`] = `
[
  "[93m[REPLACED][0m leading-1 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
  "[93m[REPLACED][0m leading-12 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
  "[93m[REPLACED][0m leading-14 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
  "[93m[REPLACED][0m leading-16 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
  "[93m[REPLACED][0m leading-20 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
  "[93m[REPLACED][0m leading-24 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
  "[93m[REPLACED][0m leading-36 -> check https://warp-ds.github.io/css-docs/line-height for supported classes",
]
`;

exports[`leading > Emits a warning if used with pseudo 2`] = `
[
  "[93m[REPLACED][0m text-primary -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[REPLACED][0m text-secondary -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
  "[93m[REPLACED][0m text-danger -> see https://warp-ds.github.io/css-docs/text-color to find suitable semantic replacement",
]
`;
